src/Formik.tsx

```diff
@@ -332,7 +329,17 @@ export class Formik<ExtraProps = {}, Values = object> extends React.Component<
   };
 
   executeSubmit = () => {
-    this.props.onSubmit(this.state.values, this.getFormikActions());
+    const schema = this.getValidationSchema();
+    const { values } = this.state;
+    const actualizedValues = schema ? schema.cast(values) : values;
+
+    this.props.onSubmit(actualizedValues, this.getFormikActions());
+  };
+
+  getValidationSchema = () => {
+    const { validationSchema } = this.props;
+
+    return isFunction(validationSchema) ? validationSchema() : validationSchema;
```

Comment:
IIRC Pass props and values to validation schema function?

Reply:
https://github.com/jaredpalmer/formik/blob/master/src/Formik.tsx#L147
Probably it's a bug in current release. Because props as arguments for validationSchema adds only here
https://github.com/jaredpalmer/formik/blob/master/src/withFormik.tsx#L127